DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1

```diff
@@ -0,0 +1,168 @@
+######################################################################################
+# DSC Resource for IIS Server level MIME Type mappings
+######################################################################################
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+NoWebAdministrationModule=Please ensure that WebAdministration module is installed.
+AddingType=Adding MIMEType '{0}' for extension '{1}'
+RemovingType=Removing MIMEType '{0}' for extension '{1}'
+TypeExists=MIMEType '{0}' for extension '{1}' already exist
+TypeNotPresent=MIMEType '{0}' for extension '{1}' is not present as requested
+TypeStatusUnknown=MIMEType '{0}' for extension '{1}' is is an unknown status
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+	param
+	(		
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$Extension,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [String]$MimeType,
+
+        [ValidateSet("Present", "Absent")]
+        [Parameter(Mandatory)]
+        [string]$Ensure
+	)
+	
+    # Check if WebAdministration module is present for IIS cmdlets
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
```

Comment:
Do we want a null case here? Given the Mandatory/NotNullOrEmpty of Set/Test shouldn't we always be returning, something?